variables.tf

```
@@ -0,0 +1,167 @@
+  }
+}
+
+variable "api_management_name" {
+  type        = string
+  description = "name of the APIM in which this diagnostic will de deployed"
+  default     = null
+  validation {
+    condition     = can(regex("^[a-zA-Z0-9-]{1,50}$", var.api_management_name))
+    error_message = "The APIM name can only contain alphanumeric characters and dashes and must be between 1 and 50 characters long."
+  }
+}
+
+variable "api_name" {
+  type        = string
+  description = "name of the API within the APIM to apply the diagnostic. when omitted, the diagnostic will be created for all APIs"
+  default     = null
+  validation {
+    condition     = var.api_name == null || can(regex("^[a-zA-Z0-9-]{1,50}$", var.api_name))
+    error_message = "The APIM name can only contain alphanumeric characters and dashes and must be between 1 and 50 characters long."
+  }
+}
+
+variable "logger_name" {
+  type        = string
+  description = "name of the logger within the APIM"
+  default     = null
+  validation {
+    condition     = can(regex("^[a-zA-Z0-9-]{1,50}$", var.logger_name))
+    error_message = "The logger name can only contain alphanumeric characters and dashes and must be between 1 and 50 characters long."
+  }
+}
+
+variable "identifier" {
+  type        = string
+  description = "Identifier of the Diagnostics Logs. Must be either 'applicationinsights' or 'azuremonitor'."
+  default     = null
+  validation {
+    condition     = can(regex("^(applicationinsights|azuremonitor)$", var.identifier))
+    error_message = "The diagnostic identifier must be either 'applicationinsights' or 'azuremonitor'."
+  }
+}
+
+# note: enabling the logging feature caused a 40%-50% reduction in throughput when request rate exceeded 1,000 requests per second
+# https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-app-insights?tabs=rest#performance-implications-and-log-sampling
+variable "sampling_percentage" {
+  type        = number
+  description = "The percentage of requests to sample. Default is '100.0'."
+  default     = 100.0
+  validation {
+    condition     = var.sampling_percentage >= 0.0 && var.sampling_percentage <= 100.0
+    error_message = "The sampling percentage must be between 0.0 and 100.0."
+  }
+}
+
+variable "always_log_errors" {
+  type        = bool
+  description = "Indicates whether to always log errors. Default is 'true'."
+  default     = true
+}
+
+variable "log_client_ip" {
+  type        = bool
+  description = "Indicates whether to log the client IP address. Default is 'false'."
+  default     = false
+}
+
+variable "verbosity" {
+  type        = string
+  description = "The verbosity level applied to the diagnostic. Must be either 'error', 'information', or 'verbose'. Default is 'error'."
+  default     = "error"
+  validation {
+    condition     = can(regex("^(error|information|verbose)$", var.verbosity))
```

Perhaps just a personal preference, but if there's no real pattern matching going on and you're just looking to test whether a value is in a small set of options, I find this easier to read:

```suggestion
    condition     = contains(["error", "information", "verbose"], var.verbosity))
```

There are a few other spots in this file where this could be applied.

---

I agree with this